### PR TITLE
Extend Event report mechanism to FEEL AST Nodes

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/EvaluationContext.java [hunk 1]
```diff
@@ -39,4 +41,8 @@
     Map<String, Object> getAllValues();
 
     FEELEventListenersManager getEventsManager();
+    
+    default void notifyEvt(Supplier<FEELEvent> event) {
```
Comment (tarilabs) [inline kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/EvaluationContext.java]: Limited default method, anyway represent a common operation which the user of the API/interface would make anyway.
Also to be noticed the `FEELEventListenersManager` is a type know from the method just above `getEventsManager()`

File: kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/BaseNode.java [hunk 2]
```diff
@@ -119,9 +124,17 @@ public String toString() {
     public Type getResultType() {
         return BuiltInType.UNKNOWN;
     }
+    
+    protected Supplier<FEELEvent> astEvent(Severity severity, String message) {
```
Comment (tarilabs) [inline kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/BaseNode.java]: Helper methods to keep it short when creating `FEELEvent`s for AST node problems

**Comment by etirelli** [comment]
@tarilabs I think the approach is good. +1

One thing we will need to do, I think, is to move all the error messages into a separate class, as a first step, so that all the messages are in a single place. Then as a second step in the future we can actually externalized them in i18n bundles. Instead of using String.format() directly in the notification calls, a factory method of that class would be called with varargs parameters to build the actual error message. What do you think?

This does not need to be done as part of this PR, but it might save some work if done sooner than later.

**Comment by tarilabs** [comment]
@etirelli agree, and indeed I would work on it with a separate PR.

**Pull request merged**